main/uart_nic.c

```diff
@@ -263,20 +247,17 @@ static int IRAM_ATTR wifi_receive_cb(void *buffer, uint16_t len, void *eb) {
         }
     }
 
-    wifi_receive_buff *buff = malloc(sizeof(wifi_receive_buff));
-    if(!buff) {
-        goto cleanup;
-    }
-    buff->len = len;
-    buff->data = buffer;
-    buff->rx_buff = eb;
-    if (!xQueueSendToBack(uart_tx_queue, (void *)&buff, (TickType_t)0/*portMAX_DELAY*/)) {
-        free_wifi_receive_buff(buff);
+    struct uart0_tx_queue_item queue_item;
+    queue_item.header.type = MSG_PACKET_V2;
+    queue_item.header.up = 1; // if we are receiving data, we surely must be "up"
```

Comment:
Around wifi, I'd be very careful about such assumptions.

I mean, I don't think anything will break if we make this assumption, but wifi is full of dark corners and ghosts…

Reply:
I can't think of anything either right now. I'm just saying that wifi is so complex and cursed technology that I wouldn't be especially surprised if we discovered some such 6 months from now and that it's worth being a bit careful about it.

So as long as we are just setting the icon to gray / white by it, it's fine to do such assumption. I wouldn't be comfortable if the printer could catch fire and burn if such packet appears. As far as I know, we just set the icon based on that, so this is any request for changes, it's just a reminder that wifi bites 😇